src/test/esperanza/finalizationstate_utils.cpp

```diff
@@ -93,16 +93,21 @@ void FinalizationStateSpy::CreateAndActivateDeposit(const uint160 &validator_add
 
   ProcessDeposit(validator_address, deposit_size);
 
-  for (uint32_t i = 1; i < 6 * EpochLength() + 1; i += EpochLength()) {
+  for (uint32_t i = 1; i < 4 * EpochLength() + 1; i += EpochLength()) {
     BOOST_REQUIRE_EQUAL(GetActiveFinalizers().size(), 0);
 
+    m_expected_source_epoch = m_last_justified_epoch;
+    m_recommended_target_epoch = m_current_epoch;
+
     res = InitializeEpoch(i);
     BOOST_REQUIRE_EQUAL(res, +Result::SUCCESS);
   }
 
-  BOOST_REQUIRE_EQUAL(GetCurrentDynasty(), 3);
-  BOOST_REQUIRE_EQUAL(GetCurrentEpoch(), 6);
-  BOOST_REQUIRE_EQUAL(GetLastJustifiedEpoch(), 4);
-  BOOST_REQUIRE_EQUAL(GetLastFinalizedEpoch(), 3);
+  BOOST_REQUIRE_EQUAL(GetCurrentDynasty(), 2);
```

Comment:
...But nothing comes after these required checks. This simply helps debugging in case it is reached to see the finalization state values at once.

Reply:
This utility function is used to build a state which will be tested further. If this function fails, it doesn't make sense to run tests based on broken state.

Reply:
I'll keep `BOOST_REQUIRE_*` as it aligned with the previous code and here I extend it. Don't want to focus on refactorings in this PR, it's better to do it separately.

Reply:
> ...But nothing comes after these required checks. This simply helps debugging in case it is reached to see the finalization state values at once.

The real tests come after these checks, in "*_tests.cpp" files. Here is a utility code that they share.